logging/log_builder_test.go

```diff
@@ -0,0 +1,148 @@
+package logging
+
+import (
+	"errors"
+
+	"github.com/mortedecai/resweave"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest/observer"
+)
+
+var _ = Describe("LogBuilder", func() {
+	var (
+		observed zapcore.Core
+		logs     *observer.ObservedLogs
+		logger   *zap.SugaredLogger
+		holder   resweave.LogHolder
+	)
+
+	Context("Log Factory", func() {
+		BeforeEach(func() {
+			observed, logs = observer.New(zapcore.DebugLevel)
+			logger = zap.New(observed).Sugar()
+			holder = resweave.NewLogholder("test", nil)
+		})
+
+		It("should be able to log info messages", func() {
+			// Arrange
+			factory := LogFactory{LogHolder: holder}
+			factory.SetLogger(logger, false)
+
+			// Act
+			factory.NewInfo("logfunc", "was-logged").Log()
+
+			Expect(logs.All()).To(HaveLen(1))
+			entry := logs.All()[0]
+			Expect(entry.Level).To(Equal(zapcore.InfoLevel))
+			Expect(entry.Message).To(Equal("logfunc"))
+			Expect(entry.ContextMap()).To(Equal(map[string]interface{}{"Status": "was-logged"}))
```

Comment:
Should still define "Status" as a constant so that you only have to change it in one place if camel casing is desired.